plugins/ingest-useragent/src/test/resources/rest-api-spec/test/ingest-useragent/20_useragent_processor.yaml

```diff
@@ -0,0 +1,87 @@
+---
+"Test user agent processor with defaults":
+  - do:
+      ingest.put_pipeline:
+        id: "my_pipeline"
+        body:  >
+          {
+            "description": "_description",
+            "processors": [
+              {
+                "useragent" : {
+                  "field" : "field1"
+                }
+              }
+            ]
+          }
+  - match: { acknowledged: true }
+
+  - do:
+      index:
+        index: test
+        type: test
+        id: 1
+        pipeline: "my_pipeline"
+        body: {field1: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/33.0.1750.149 Safari/537.36"}
+
+  - do:
+      get:
+        index: test
+        type: test
+        id: 1
+  - match: { _source.field1: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/33.0.1750.149 Safari/537.36" }
+  - match: { _source.useragent.name: "Chrome" }
+  - match: { _source.useragent.os: "Mac OS X 10.9.2" }
+  - match: { _source.useragent.os_name: "Mac OS X" }
+  - match: { _source.useragent.os_major: "10" }
+  - match: { _source.useragent.os_minor: "9" }
+  - match: { _source.useragent.major: "33" }
+  - match: { _source.useragent.minor: "0" }
+  - match: { _source.useragent.patch: "1750" }
+  - match: { _source.useragent.device: "Other" }
+
+---
+"Test user agent processor with parameters":
+  - do:
+      ingest.put_pipeline:
+        id: "my_pipeline"
+        body:  >
+          {
+            "description": "_description",
+            "processors": [
+              {
+                "useragent" : {
+                  "field" : "field1",
+                  "target_field": "field2",
+                  "cache_size": 0,
```

Comment:
That is because `gradle test` only runs unit tests and this is an integration test. `gradle integTest` only runs integration tests and if you run `gradle check` both unit and integration tests are ran and some additional things like checking for forbidden apis and if the code adheres to the code style.